### ISSUE TITLE
Need https to use the microphone from website

### ISSUE THREAD

**Issue by mbuccoli** [hajicj/HAMR_2016#6]
Hello, I've just tried to put an online version of our demo at
http://home.deib.polimi.it/buccoli/RAP

But I get this error message

> recorder.js:98 getUserMedia() no longer works on insecure origins. To use this feature, you should consider switching your application to a secure origin, such as HTTPS. See https://goo.gl/rStTGz for more details.

If I use https://home.deib.polimi.it/buccoli/RAP instead, I do not receive the error, but when I try to rap I get the message

> jquery-1.11.1.min.js:4 Mixed Content: The page at 'https://home.deib.polimi.it/buccoli/RAP/' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://mir.deib.polimi.it:8001/'. This request has been blocked; the content must be served over HTTPS.send @ jquery-1.11.1.min.js:4ajax @ jquery-1.11.1.min.js:4sendBlob @ recorder.js:50Recorder.worker.onmessage @ recorder.js:343
> jquery-1.11.1.min.js:4 XMLHttpRequest cannot load http://mir.deib.polimi.it:8001/. Failed to start loading.send @ jquery-1.11.1.min.js:4ajax @ jquery-1.11.1.min.js:4sendBlob @ recorder.js:50Recorder.worker.onmessage @ recorder.js:343

So basically we need to figure it out how to setup a secure http server from python. Any clue?

**Comment by MECHMulimedia** [comment]
localhost is treated as a secure origin over HTTP, so if you're able to run your server from localhost, you should be able to test the feature on that server.

**Comment by mbuccoli** [comment]
Yes, indeed I'm not having issues at all with the localhost version, I'm making an attempt to run a real version of the demo.

**Comment by mbuccoli** [comment]
Ok, basically it is working but we need to register an SSL certificate to make it properly work on the Internet.
You can try it by first visiting the page
https://mir.deib.polimi.it:8001
your browser (usually chrome) will tell you the page is not secure (since it does not have a public key registered to the authority). Just choose the option to go on and download the certificate.

After it, you can just visit
http://home.deib.polimi.it/buccoli/RAP
and start trying your raps.
